src/ctap/ctap1.rs

```diff
@@ -180,15 +178,12 @@ impl Ctap1Command {
     const VENDOR_SPECIFIC_FIRST: u8 = 0x40;
     const VENDOR_SPECIFIC_LAST: u8 = 0xBF;
 
-    pub fn process_command<R, CheckUserPresence>(
+    pub fn process_command(
+        env: &mut impl Env,
         message: &[u8],
-        ctap_state: &mut CtapState<R, CheckUserPresence>,
+        ctap_state: &mut CtapState,
```

Comment:
This function takes a CtapState, just like the `process_*` in `hid` does. I'm thinking this should eventually move. Unlike all other commands (MakeCredential etc.), this function is actually called directly from HID.

Reply:
Yes I was surprised to see those functions take a `CtapState`. I didn't try to see why yet, but that's indeed something that could be addressed during refactoring.

Reply:
I think it's just the hierarchy of our current design. `CtapState` is to `hid` what `check_user_presence` is for `ctap`. The packet gets processed from receive to send, and the CTAP logic is called in between.

Reply:
The above was the answer for HID, for CTAP1, the `CtapState` mostly takes the role of `env`, but there is also `decrypt_credential_source` (and encrypt). These functions both don't depend on `CtapState`, but on `env` themselves, so the new design should seperate them.